source/text_handler.cpp

```diff
@@ -104,35 +104,29 @@ spv_result_t advance(spv_text text, spv_position position) {
   return SPV_SUCCESS;
 }
 
-/// @brief Fetch the next word from the text stream.
-///
-/// A word ends at the next comment or whitespace.  However, double-quoted
-/// strings remain intact, and a backslash always escapes the next character.
-///
-/// @param[in] text stream to read from
-/// @param[in] position current position in text stream
-/// @param[out] word returned word
-/// @param[out] endPosition one past the end of the returned word
-///
-/// @return result code
-spv_result_t getWord(spv_text text, spv_position position, std::string& word,
-                     spv_position endPosition) {
+// Fetches the next word from the given text stream starting from the given
+// *position. On success, writes the decoded word into *word and updates
+// *position to the location past the returned word.
+//
+// A word ends at the next comment or whitespace.  However, double-quoted
+// strings remain intact, and a backslash always escapes the next character.
+spv_result_t getWord(spv_text text, spv_position position, std::string* word) {
   if (!text->str || !text->length) return SPV_ERROR_INVALID_TEXT;
-  if (!position || !endPosition) return SPV_ERROR_INVALID_POINTER;
+  if (!position) return SPV_ERROR_INVALID_POINTER;
 
-  *endPosition = *position;
+  const size_t start_index = position->index;
 
   bool quoting = false;
   bool escaping = false;
 
   // NOTE: Assumes first character is not white space!
   while (true) {
-    if (endPosition->index >= text->length) {
-      word.assign(text->str + position->index,
-                  static_cast<size_t>(endPosition->index - position->index));
+    if (position->index >= text->length) {
+      word->assign(text->str + start_index,
+                   static_cast<size_t>(position->index - start_index));
```

Comment:
Optional: `assign(text->str + start_index, text->str + position->index)` is shorter and more readable.